skyportal/models.py

```diff
@@ -3448,6 +3454,334 @@ class SourceNotification(Base):
     level = sa.Column(sa.String(), nullable=False)
 
 
+class GcnNotice(Base):
```

Comment:
Another thing that we should probably think about when it comes to these model changes is specifying access control rules. That is, specifying which users have permissions to read these records, who's allowed to delete or update them, etc. You can read more about how to do it here: https://skyportal.io/docs/permissions.html. But yeah, generally I'm looking for things like `delete = restricted` and `read = public`, stuff like that.

Reply:
Yes I added what I thought made sense. Basically I want these all public, but maybe best not to have just anyone delete them. Can you confirm that I got it right for that?

Reply:
Yup makes sense. In order for these to work properly, you'll need to define the `sent_by` relationships to the `User` table so that the records actually know who sent that data in, and you'll also need to update the `GCNHandler` to store that information. A simple example you could look to would be the `Listing` table, which defines rules against the `user` that the listing entry is for.

Reply:
@kmshin1397 This piece you mean?

User.listings = relationship(
    'Listing',
    back_populates='user',
    passive_deletes=True,
    doc='The listings saved by this user',
)